src/components/MultiHopTrade/components/LimitOrder/components/LimitOrderConfirm.tsx

```diff
@@ -10,36 +10,103 @@ import {
 import { useHistory } from 'react-router'
-import { ethereum, fox } from 'test/mocks/assets'
 import { Amount } from 'components/Amount/Amount'
 import { AssetToAssetCard } from 'components/AssetToAssetCard/AssetToAssetCard'
 import { Row } from 'components/Row/Row'
 import { SlideTransition } from 'components/SlideTransition'
 import { RawText, Text } from 'components/Text'
 import { TransactionDate } from 'components/TransactionHistoryRows/TransactionDate'
+import { useActions } from 'hooks/useActions'
+import { useErrorHandler } from 'hooks/useErrorToast/useErrorToast'
+import { useWallet } from 'hooks/useWallet/useWallet'
+import { usePlaceLimitOrderMutation } from 'state/apis/limit-orders/limitOrderApi'
+import { limitOrderSlice } from 'state/slices/limitOrderSlice/limitOrderSlice'
+import {
+  selectActiveQuote,
+  selectActiveQuoteBuyAmountCryptoPrecision,
+  selectActiveQuoteBuyAmountUserCurrency,
+  selectActiveQuoteBuyAsset,
+  selectActiveQuoteExpirationTimestamp,
+  selectActiveQuoteFeeAsset,
+  selectActiveQuoteLimitPrice,
+  selectActiveQuoteNetworkFeeCryptoPrecision,
+  selectActiveQuoteSellAmountCryptoPrecision,
+  selectActiveQuoteSellAmountUserCurrency,
+  selectActiveQuoteSellAsset,
+} from 'state/slices/limitOrderSlice/selectors'
+import { useAppSelector } from 'state/store'
 
 import { SwapperIcon } from '../../TradeInput/components/SwapperIcon/SwapperIcon'
 import { WithBackButton } from '../../WithBackButton'
 import { LimitOrderRoutePaths } from '../types'
 
 const cardBorderRadius = { base: '2xl' }
+
+// TODO: Populate this!
 const learnMoreUrl = ''
 
 export const LimitOrderConfirm = () => {
   const history = useHistory()
   const translate = useTranslate()
+  const wallet = useWallet().state.wallet
+  const { confirmSubmit, setLimitOrderInitialized } = useActions(limitOrderSlice.actions)
+  const { showErrorToast } = useErrorHandler()
+
+  const activeQuote = useAppSelector(selectActiveQuote)
+  const sellAsset = useAppSelector(selectActiveQuoteSellAsset)
+  const buyAsset = useAppSelector(selectActiveQuoteBuyAsset)
+  const feeAsset = useAppSelector(selectActiveQuoteFeeAsset)
+  const sellAmountCryptoPrecision = useAppSelector(selectActiveQuoteSellAmountCryptoPrecision)
+  const buyAmountCryptoPrecision = useAppSelector(selectActiveQuoteBuyAmountCryptoPrecision)
+  const sellAmountUserCurrency = useAppSelector(selectActiveQuoteSellAmountUserCurrency)
+  const buyAmountUserCurrency = useAppSelector(selectActiveQuoteBuyAmountUserCurrency)
+  const networkFeeCryptoPrecision = useAppSelector(selectActiveQuoteNetworkFeeCryptoPrecision)
+  const limitPrice = useAppSelector(selectActiveQuoteLimitPrice)
+  const quoteExpirationTimestamp = useAppSelector(selectActiveQuoteExpirationTimestamp)
+
+  const [placeLimitOrder, { data, error, isLoading }] = usePlaceLimitOrderMutation()
+
+  useEffect(() => {
+    if (!error) return
+
+    const description = (error as CowSwapError).description ?? 'Unknown Error'
```

Comment:
q: can't we directly return a `CowSwapError` from the mutation?

Reply:
We do, but as the api may actually also can return a `SerializedError` which is internal to RTK query, we will need to handle it, treating it as "unknown"